spring-boot-actuator/src/test/java/org/springframework/boot/actuate/endpoint/mvc/SessionMvcEndpointTests.java

```diff
@@ -0,0 +1,112 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.endpoint.mvc;
+
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.autoconfigure.EndpointWebMvcAutoConfiguration;
+import org.springframework.boot.actuate.autoconfigure.ManagementServerPropertiesAutoConfiguration;
+import org.springframework.boot.autoconfigure.session.SessionAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Tests for {@link SessionMvcEndpoint}
+ *
+ * @author Eddú Meléndez
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class SessionMvcEndpointTests {
+
+	@Autowired
+	private MvcEndpoints endpoints;
+
+	@Autowired
+	private WebApplicationContext context;
+
+	private MockMvc mvc;
+
+	@Before
+	public void setUp() {
+		this.mvc = MockMvcBuilders.webAppContextSetup(this.context).build();
+	}
+
+	@Test
+	public void endpointRegistered() {
+		Set<? extends MvcEndpoint> values = this.endpoints.getEndpoints();
+		assertThat(values).hasAtLeastOneElementOfType(SessionMvcEndpoint.class);
+	}
+
+	@Test
+	public void searchEndpointAvailable() throws Exception {
+		this.mvc.perform(get("/session/springuser"))
+				.andExpect(status().isOk());
+	}
+
+	@Test
+	public void deleteEndpointAvailable() throws Exception {
+		this.mvc.perform(delete("/session/123"))
+				.andExpect(status().isOk());
+	}
+
+	@Configuration
+	@EnableConfigurationProperties
+	@EnableWebMvc
+	@Import({ TestRedisConfiguration.class, SessionAutoConfiguration.class,
```

Comment:
I wonder if it makes more sense to use a mock of `FindByIndexNameSessionRepository`